src/tunnel.cc

```diff
@@ -625,6 +661,20 @@ TunnelStateData::Connection::write(const char *b, int size, AsyncCall::Pointer &
     Comm::Write(conn, b, size, callback, free_func);
 }
 
+void
+TunnelStateData::Connection::setCloseHandler(CLCB *handler, TunnelStateData *tunnelState)
+{
+    assert(handler);
+    assert(tunnelState);
+    if (!Comm::IsConnOpen(conn)) {
+        closer = nullptr;
+        return;
+    }
+    if (closer)
```

Comment:
If we may be resetting an old handler, then let's rename this method into resetCloseHandler() to reduce caller surprises.